examples/spanner_grpcgcp/spanner_grpcgcp.go

```diff
@@ -0,0 +1,211 @@
+					"/google.spanner.v1.Spanner/Rollback",
+					"/google.spanner.v1.Spanner/StreamingRead",
+				},
+				Affinity: &gpb.AffinityConfig{
+					Command:     gpb.AffinityConfig_BOUND,
+					AffinityKey: "session",
+				},
+			},
+		},
+	}
+)
+
+// ConnPool wrapper for gRPC-GCP channel pool. gtransport.ConnPool is the
+// interface Spanner client accepts as a replacement channel pool.
+type grpcGcpConnPool struct {
+	gtransport.ConnPool
+
+	cc   *grpc.ClientConn
+	size int
+}
+
+func (cp *grpcGcpConnPool) Conn() *grpc.ClientConn {
+	return cp.cc
+}
+
+// Spanner client uses this function to get channel pool size.
+func (cp *grpcGcpConnPool) Num() int {
+	return cp.size
+}
+
+func (cp *grpcGcpConnPool) Close() error {
+	return cp.cc.Close()
+}
+
+func databaseURI() string {
+	return fmt.Sprintf("projects/%s/instances/%s/databases/%s", *project, *instance_name, *database_name)
+}
+
+func spannerClient() (*spanner.Client, error) {
+	var perRPC credentials.PerRPCCredentials
+	var err error
+	keyFile := os.Getenv("GOOGLE_APPLICATION_CREDENTIALS")
+	if keyFile == "" {
+		perRPC, err = oauth.NewApplicationDefault(context.Background(), scope)
+	} else {
+		perRPC, err = oauth.NewServiceAccountFromFile(keyFile, scope)
+	}
+
+	// Converting gRPC-GCP config to JSON because grpc.Dial only accepts JSON
+	// config for configuring load balancers.
+	grpcGcpJsonConfig, err := protojson.Marshal(grpcGcpConfig)
+	if err != nil {
+		return nil, err
+	}
+
+	conn, err := grpc.Dial(
+		// Spanner endpoint. Replace this with your custom endpoint if not using
+		// the default endpoint.
+		endpoint,
+		// Application default or service account credentials set up above.
+		grpc.WithTransportCredentials(credentials.NewClientTLSFromCert(nil, "")),
+		grpc.WithPerRPCCredentials(perRPC),
+		// Do not look up load balancer (gRPC-GCP) config via DNS.
+		grpc.WithDisableServiceConfig(),
+		// Instead use this static config.
+		grpc.WithDefaultServiceConfig(fmt.Sprintf(`{"loadBalancingConfig": [{"%s":%s}]}`, grpcgcp.Name, string(grpcGcpJsonConfig))),
+		// gRPC-GCP interceptors required for proper operation of gRPC-GCP
+		// channel pool. Add your interceptors as next arguments if needed.
+		grpc.WithChainUnaryInterceptor(grpcgcp.GCPUnaryClientInterceptor),
+		grpc.WithChainStreamInterceptor(grpcgcp.GCPStreamClientInterceptor),
+		// Add more DialOption options if needed but make sure not to overwrite
+		// the interceptors above.
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	pool := &grpcGcpConnPool{
+		cc: conn,
+		// Set the pool size on ConnPool to communicate it to Spanner client.
+		size: poolSize,
+	}
+
+	// Create spanner client. gtransport.WithConnPool is one of the available
+	// Spanner client options. Keep other Spanner-related options here if any
+	// and move channel pool and dial option to the grpc.Dial above.
+	return spanner.NewClient(context.Background(), databaseURI(), gtransport.WithConnPool(pool))
+}
+
+func main() {
+	flag.Parse()
```

Comment:
Maybe validate inputs?

Reply:
It will print an error that it couldn't connect and will show a string with missing project/instance/database. I think this is good enough for an example.